spring-cloud-gcp-autoconfigure/src/test/java/com/google/cloud/spring/autoconfigure/pubsub/it/PubSubTemplateDocumentationIntegrationTests.java

```diff
@@ -93,6 +95,35 @@ public void testCreatePublishPullNextAndDelete() {
 		});
 	}
 
+	@Test
+	public void testCreatePublishPullNextAndDelete_ordering() {
+		pubSubTest((AssertableApplicationContext context, PubSubTemplate pubSubTemplate, String subscriptionName, String topicName) -> {
+			//tag::publish_ordering[]
+			Map<String, String> headers = Collections.singletonMap(GcpPubSubHeaders.ORDERING_KEY, "key1");
+			pubSubTemplate.publish(topicName, "message1", headers).get();
+			pubSubTemplate.publish(topicName, "message2", headers).get();
```

Comment:
Maybe add a comment to explain that messages with the same ordering key will be delivered in the same order they were sent.